### DISCUSSION TITLE
[is this possible?]"Storage Charge" mode for built-in charger

### DISCUSSION THREAD

**Discussion by Dweinbach** [EdgeTX/edgetx#1181] [Ideas]
It would be handy to "storage charge" the onboard batteries to 3.84 or 3.85 v/cell in the Radiomaster TX16S and possibly other radios with built in chargers.

Not sure if the circuitry allows this level of control, but it would be handy to storage charge the batteries.

As a minimum, stop charging when cells reach storage levels. extra-credit for figuring out what to turn on to discharge to storage level without burning up the radio or getting an FCC SWAT team dispatched!

**Comment by raphaelcoeffic** [comment]
Sorry, we have zero control over the charging circuits. They are as dumb as possible. I know we are now used to phone that can do all that, but that is not the case here.

**Comment by pfeerick** [comment]
As Raphael said, the transmitter has zero control over the charging of the batteries, as that is handled by dedicated charge management circuitry. Your only option at present will be to put the batteries onto a conventional hobby charge that supports storage (dis)charging the batteries.

**Comment by Dweinbach** [comment]
I suspected as much, but thought maybe it was cheaper for them to program the charging circuit on the fly using processor I/O than to hard-wire the target voltage, charging current, etc.

Oh well. Thanks!

**Comment by Dweinbach** [comment]
I came across this old discussion again, and I realized that as a minimum, I could make a SF that makes some noise when the battery voltage reaches 7.7. What is the best way to detect that I am NOT connected to a model in a SF, so it will only sound the "storage charge reached" alarm when I am NOT flying (using internal multiprotocol model and DSM protocol)?

**Reply by pfeerick** [reply]
If you have telemetry, perhaps monitor for the absence of that. Otherwise, you can probably really only do something like monitor the throttle position. Better still would perhaps be a Lua widget, as then you have a lot more flexibility over the logic and can do conditional tests better. Plus it would be easy transferable between models.
